docs/docs/v3-adoption-progress.md

```diff
@@ -0,0 +1,155 @@
+---
+title: "Apache Iceberg V3 Table Format Adoption Progress"
+---
+<!--
+ - Licensed to the Apache Software Foundation (ASF) under one or more
+ - contributor license agreements.  See the NOTICE file distributed with
+ - this work for additional information regarding copyright ownership.
+ - The ASF licenses this file to You under the Apache License, Version 2.0
+ - (the "License"); you may not use this file except in compliance with
+ - the License.  You may obtain a copy of the License at
+ -
+ -   http://www.apache.org/licenses/LICENSE-2.0
+ -
+ - Unless required by applicable law or agreed to in writing, software
+ - distributed under the License is distributed on an "AS IS" BASIS,
+ - WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ - See the License for the specific language governing permissions and
+ - limitations under the License.
+ -->
+
+# Apache Iceberg V3 Table Format Adoption Progress
+
+This document tracks the implementation status of Apache Iceberg V3 Table Format features across different engines and components.
+
+## Overview
+
+Apache Iceberg V3 Table Format introduces several new features including new data types, row lineage, deletion vectors, default values, and table encryption. This document provides a comprehensive comparison of V3 feature implementation status across Apache Iceberg Core, Apache Spark, and Apache Flink.
+
+## V3 Features Implementation Status
+
+| **V3 Feature** | **Core** | **Spark** | **Flink** | **Notes** |
+|---|---|---|---|---|
+| **New Data Types** | | | | |
+| `unknown` type | ✅ **Implemented** | ⚠️ **Partially Implemented** | ✅ **Implemented** | Core: Full support. Spark: v4.0+ ✅, v3.4/v3.5 ❌. Flink: Full support. |
+| `timestamp_ns` (nanosecond timestamps) | ✅ **Implemented** | ❌ **Not Implemented** | ✅ **Implemented** | Core: Full support including timestamptz_ns. Spark: Not supported in any version. Flink: Full support including timestamptz_ns. |
+| `variant` type | ⚠️ **Partially Implemented** | ❌ **Not Implemented** | ❌ **Not Implemented** | Core: Avro ✅, Parquet ✅, ORC ❌. Spark/Flink: No support. |
```

Comment:
For variant type,  Spark is Partially Implemented - we are missing writing shredded variant but we can read both shredded and unshredded variant in Spark.